lib/resolve-value.js

```diff
@@ -27,26 +29,39 @@ function toString(value) {
 var resolveValue = function(decl, map, /*optional*/ignorePseudoScope, /*internal debugging*/_debugIsInternal) {
 	var debugIndent = _debugIsInternal ? '\t' : '';
 
+	var matchingVarDecl = undefined;
+	var RE_VAR_FUNC_G = new RegExp(RE_VAR_FUNC.source, 'g');
 	var resultantValue = toString(decl.value);
 	var warnings = [];
+	
 
-	var variablesUsedInValueMap = {};
-	// Use `replace` as a loop to go over all occurrences with the `g` flag
-	resultantValue.replace(new RegExp(RE_VAR_FUNC.source, 'g'), function(match, variableName, fallback) {
-		variablesUsedInValueMap[variableName] = true;
-	});
-	var variablesUsedInValue = Object.keys(variablesUsedInValueMap);
+	// match all variables first, to gather all available variables in decl.value
+	var variablesUsedInValue = [];
+	while ((matchingVarDecl = RE_VAR_FUNC_G.exec(resultantValue))) {
```

Comment:
Why are we using a different matching strategy from below? Seems like we should use the same strategy to get the `variableName` as below

Reply:
Reason: I ended up maintaining this strategy, to give more purpose to `RE_VAR_FUNC` RegExp.

Check it now, I changed the matching strategy, to use `balanced` npm lib as well.

Reply:
Seems good now. Both use the same looping strategy